android/src/main/java/cl/json/social/InstagramShare.java

```diff
@@ -24,7 +24,15 @@ public InstagramShare(ReactApplicationContext reactContext) {
     @Override
     public void open(ReadableMap options) throws ActivityNotFoundException {
         super.open(options);
-        //  extra params here
+        try {
+            // passing instagram://share as url option opens Camera View for publishing
+            if (ShareIntent.hasValidKey("url", options) && options.getString("url").startsWith("instagram://")) {
+                this.getIntent().setAction(Intent.ACTION_VIEW);
+                this.getIntent().setData(Uri.parse(options.getString("url")));
+            }
+        }catch (Exception e){
+            // bad param passed
```

Comment:
If these are simply consumed with no output whatsover, when the params required inevitably change upstream there will be no notice for anyone anywhere. So these should be logged somehow. The question is severity. For severity "error" means execution should halt as state is undefined - that is not correct here, "info" means no problem really just something you should know - that is also not quite right, "warning" means something unexpected and bad but we think we can continue. I think that fits.

```suggestion
            // bad param passed
            android.util.Log.warn("RNShare", e.getMessage());
```